dask/dataframe/io/parquet.py

```diff
@@ -693,78 +665,58 @@ def _read_pyarrow(fs, fs_token, paths, columns=None, filters=None,
 
-    # Find non-empty pieces
-    non_empty_pieces = []
-    # Determine valid pieces
-    _open = lambda fn: pq.ParquetFile(fs.open(fn, mode='rb'))
-    for piece in dataset.pieces:
-        pf = piece.get_metadata(_open)
-        # non_empty_pieces.append(piece)
-        if pf.num_row_groups > 0:
-            non_empty_pieces.append(piece)
-
-    # Sort pieces naturally
-    # If a single input path resulted in multiple dataset pieces, then sort
-    # the pieces naturally. If multiple paths were supplied then we leave
-    # the order of the resulting pieces unmodified
-    if len(paths) == 1 and len(dataset.pieces) > 1:
-        non_empty_pieces = sorted(
-            non_empty_pieces, key=lambda piece: natural_sort_key(piece.path))
-
-    # Determine divisions
-    if len(index_names) == 1:
-
-        # Look up storage name of the single index column
-        divisions_names = [storage_name for storage_name, name
-                           in storage_name_mapping.items()
-                           if index_names[0] == name]
-
-        if divisions_names:
-            divisions_name = divisions_names[0]
-        else:
-            divisions_name = None
-    else:
-        divisions_name = None
-
-    divisions = _get_pyarrow_divisions(non_empty_pieces, divisions_name,
-                                       schema, infer_divisions)
+    pieces = sorted(dataset.pieces, key=lambda piece: natural_sort_key(piece.path))
 
-    # Build task
     dtypes = _get_pyarrow_dtypes(schema, categories)
     dtypes = {storage_name_mapping.get(k, k): v for k, v in dtypes.items()}
 
     meta = _meta_from_dtypes(all_columns, dtypes, index_names,
                              column_index_names)
     meta = clear_known_categories(meta, cols=categories)
 
-    if out_type == Series:
-        assert len(meta.columns) == 1
-        meta = meta[meta.columns[0]]
-
-    task_name = 'read-parquet-' + tokenize(fs_token, paths, all_columns)
-
-    if non_empty_pieces:
-        task_plan = {
-            (task_name, i): (_read_pyarrow_parquet_piece,
-                             fs,
-                             piece,
-                             column_names,
-                             index_names,
-                             out_type == Series,
-                             dataset.partitions,
-                             categories)
-            for i, piece in enumerate(non_empty_pieces)
-        }
+    if gather_statistics:
+        # Read from _metadata file
+        if dataset.metadata and dataset.metadata.num_row_groups == len(dataset.pieces):  # one rg per piece
+            row_groups = [
+                dataset.metadata.row_group(i)
+                for i in range(dataset.metadata.num_row_groups)
+            ]
+        # Read from each individual piece (possibly slow)
+        else:
+            row_groups = [
+                piece.get_metadata(lambda fn: pq.ParquetFile(fs.open(fn, mode='rb'))).row_group(0)
```

Comment:
Yes.  I'm not sure if we actually want to do this.  My guess is that we want to do something like gather statistics if there is a metadata file, or maybe if we're using a local file system, but otherwise don't unless the user explicitly tells us to.

Reply:
Maybe could have a timer that warns if the process is taking longer than some limit?

Reply:
I propose a `statistics=None` keyword that can be 

-  True: get statistics, even if no metadata file is present
-  False: don't get statistics, even if a metadata file is present
-  None: get statistics only if a metadata file is present

In this ways users need to explicitly opt-in to expensive behavior.

Reply:
If we start focusing on Arrow and adopt a policy like this then it would be good to be able to write metadata files with arrow.  Reported upstream in [Arrow-1983](https://issues.apache.org/jira/browse/ARROW-1983).

Reply:
cc @wesm